oio/xcute/modules/blob_mover.py

```diff
@@ -87,65 +91,70 @@ def process(self, chunk_url, rawx_timeout=None, min_chunk_size=None,
             chunk_id, fake_excluded_chunks=fake_excluded_chunks, **kwargs)
 
         self.logger.info('Moved chunk %s to %s', chunk_url, new_chunk['url'])
+        return chunk_size
 
 
-class RawxDecommissionJob(XcuteJob):
+class RawxDecommissionModule(XcuteModule):
 
-    JOB_TYPE = 'rawx-decommission'
+    MODULE_TYPE = 'rawx-decommission'
     DEFAULT_RDIR_FETCH_LIMIT = 1000
     DEFAULT_RDIR_TIMEOUT = 60.0
     DEFAULT_RAWX_TIMEOUT = 60.0
     DEFAULT_MIN_CHUNK_SIZE = 0
     DEFAULT_MAX_CHUNK_SIZE = 0
     DEFAULT_EXCLUDED_RAWX = list()
 
-    def __init__(self, conf, job_info, create=False, logger=None):
-        super(RawxDecommissionJob, self).__init__(
-            conf, job_info, create=create, logger=logger)
+    def __init__(self, conf, options, details, logger=None):
+        super(RawxDecommissionModule, self).__init__(
+            conf, options, details, logger=logger)
 
-        self.rdir_client = RdirClient(self.conf, logger=self.logger)
-
-    def _parse_job_info(self, create=False):
-        super(RawxDecommissionJob, self)._parse_job_info(create=create)
-
-        job_config = self.job_info.setdefault('config', dict())
-        self.service_id = job_config.get('service_id')
-        if not self.service_id:
-            raise ValueError('Missing service ID')
+        self.rawx_id = options.get('rawx_id')
+        if not self.rawx_id:
+            raise ValueError('Missing rawx ID')
```

Comment:
all of this needs to be validated before we start the job, before we even create it